enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,882 @@
+inside the container for managing the kernel module (dkms like functions).
+
+The following section will walk one through the enablement of the different
+use-case scenarios. After deploying the operator the first step is to create an
+instance of a special-resource. Following are some example CRs how to one would
+instantiate SROv2 to manage a kernel module or hardware driver.
+
+#### Example CR for a single kernel module #1
+
+```yaml
+apiVersion: sro.openshift.io/v1alpha1
+kind: SpecialResource
+metadata:
+  name: <vendor>-<kmod>
+spec:
+  driverContainer:
+  - git:
+      ref: "release-4.3"
+      uri: "https://gitlab.com/<vendor>/<kmod>.git"
+```
+
+The second example shows the combined capabilities of SROv2 for dealing with
+multiple driver containers and artifacts. On the other side SROv2 can also be
+used in a minimalistic form where we only deploy a simple kmod. The example CR
+above would create only one DriverContainer from the git repository provided.
+For each kernel module one would provide one CR with the needed information.
+
+#### Example CR for a hardware vendor (all settings) #2
+
+```yaml
+apiVersion: sro.openshift.io/v1alpha1
+kind: SpecialResource
+metadata:
+  name: <vendor>-<hardware>
+spec:
+  metadata:
+    namespace: <vendor>-<driver>
+  environment:
+  - key: "key_id"
+    value: "ACCESS_KEY_ID"
+  - key: "access_key"
+    value: "SECRET_ACCESS_KEY"
+  driverContainer:
+    source:
+      git:
+        ref: "master"
+        uri: "https://gitlab.com/<vendor>/driver.git"
+    buildArgs:
+    - name: "DRIVER_VERSION"
+      value: "440.64.00"
+    - name: "USE_SPECIFIC_DRIVER_FEATURE"
+      value: "True"
+    runArgs:
+    - name: "LINK_TYPE_P1" # 1st Port
+      value: "2"  #Ethernet
+    - name: "LINK_TYPE_P2" # 2nd Port
+      value: "2"  #Ethernet
+    artifacts:
+      hostPaths:
+      - sourcePath: "/run/<vendor>/usr/src/<artifact>"
+        destinationDir: "/usr/src/"
+      images:
+      - name: "<vendor>-{{.KernelVersion}}:latest"
+        kind: ImageStreamTag
+        namespace: "<vendor>-<hardware>"
+        pullSecret: "vendor-secret"
+        paths:
+        - sourcePath: "/usr/src/<vendor>/<artifact>
+          destinationDir: "/usr/src/"
+      claims:
+      - name: "<vendor>-pvc"
+      mountPath: "/usr/src/<vendor>-<internal>"
+  node:
+    selector: "feature.../pci-<VENDOR_ID>.present"
+  dependsOn:
+    - name: <CR_NAME_VENDOR_ID_SRO>
+    - name: <CR_NAME_VENDOR_ID_KJI>
+```
+
+Since SROv2 will manage several special resources in different namespaces, hence
+the CRD will have cluster scope. The SROv2 can take care of creating and
+deleting of the namespace for the specialresource. Otherwise one would have an
+manual step in creating the new namespace before creating the CR for a
+specialresource, which make cleanup of a special resource easy, just by deleting
+the namespace. If there is no spec.metadata.namespace supplied SROv2 will set
```

Comment:
```suggestion
the CRD will have cluster scope. The SROv2 can take care of creating and 
deleting of the namespace for the specialresource, which makes cleanup of
a special resource easy, just by deleting the namespace. Otherwise one would have a 
manual step in creating the new namespace before creating the CR for a specialresource.  
If there is no spec.metadata.namespace supplied SROv2 will set
```